### ISSUE TITLE
Create plugin fails if the name of the plugin matches to an existing /tmp subfolder

### ISSUE THREAD

**Issue by adamdmharvey** [spotify/backstage#3164]
When creating a new plugin, if the name of the plugin happens to conflict with the name of an existing subfolder of `/tmp`, the plugin creation fails.

## Expected Behavior

The plugin should be able to be created without conflicting with existing files in the `/tmp` folder that may not have anything at all to do with the Backstage plugin creation.

## Current Behavior

If the name of a chosen new plugin matches exactly one subfolder of `/tmp`, then the creation fails.

e.g., if `/tmp/foo` exists, you can't create a new pluging called `foo`.

## Possible Solution

Prefix some sort of temporary name (`temp-plugin-`), or randomized number, and use that for the temporary directory in use while generating the plugin.

## Steps to Reproduce

1. Clone a raw copy of `spotify/backstage`
2. `yarn install` - install code
3. `mkdir /tmp/foo` - create a preexisting "foo" directory
4. `yarn create-plugin`
5. Name your new plugin: "foo"
6. See error.

## Context

```sh
user@hostname:/tmp$ mkdir randomo
user@hostname:/tmp$ cd /mnt/c/github/tempfoo/backstage/
user@hostname:/mnt/c/github/tempfoo/backstage(master)$ yarn create-plugin
yarn run v1.22.1
$ backstage-cli create-plugin --scope backstage --no-private
? Enter an ID for the plugin [required] randomo
? Enter the owner(s) of the plugin. If specified, this will be added to CODEOWNERS for the plugin path. [optional] rando
m@o.com

Creating the plugin...

 Checking if the plugin ID is available:
  checking      /mnt/c/github/tempfoo/backstage/plugins/randomo ✔

 Creating a temporary plugin directory:
  creating      temporary directory ✖

Failed to create temporary plugin directory: EEXIST: file already exists, mkdir '/tmp/randomo'

It seems that something went wrong when creating the plugin 🤔
We are going to clean up, and then you can try again.

 Cleanup:
  remove        temporary directory ✔

🔥  Failed to create plugin!

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

## Your Environment

- NodeJS Version (v12):  v12.16.2
- Operating System and Version (e.g. Ubuntu 14.04): Ubuntu 18.04 on Windows 10 1909 and WSL1
- Browser Information: Microsoft Edge (Chrome)

**Comment by adamdmharvey** [comment]
PS: a side effect, I noticed, is if there are actual files in your existing `/tmp` directory, than not only will this fail but it removes the temp directory.  That should be OK, unless some other app is actively still using those temp files in another process. So by avoiding the failure to create an existing folder, it should also avoid removing one it shouldn't remove!

**Comment by Rugvip** [comment]
Thank you for reporting! Went straight for a fix with #3165
